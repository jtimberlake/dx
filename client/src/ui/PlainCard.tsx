import React, { FC } from 'react';
import styled from 'styled-components';
import { CardBase } from '../ui/Card';
import { theme, Color } from '../theme';

const Card = styled(CardBase)`
  padding: ${theme.spacing.unit * 2}px;
`;
const Header = styled.h2`
  color: ${Color['neutral-550']};
  font-weight: normal;
  font-size: ${theme.fontSize[18]};
  margin: 0 0 ${theme.spacing.unit * 2}px 0;
`;

const PlainCard: FC<{ title: string }> = ({ title, children, ...props }) => {
  return (
    <Card {...props}>
      <Header>{title}</Header>
      {children}
    </Card>
  );
};

export default PlainCard;
