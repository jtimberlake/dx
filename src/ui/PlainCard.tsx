import React, { FC } from 'react';
import styled from 'styled-components';
import { Card, CardContent, CardFooter } from '../ui/Card';
import { theme, Color } from '../theme';

const Header = styled.h2`
  color: ${Color['neutral-550']};
  font-weight: normal;
  font-size: ${theme.fontSize[18]};
  margin: 0 0 ${theme.spacing.unit * 2}px 0;
`;

const PlainCard: FC<{ title?: string }> = ({ title, children, ...props }) => {
  return (
    <Card collapsing={false}>
      {title && <Header>{title}</Header>}
      <CardContent>{children}</CardContent>
      <CardFooter></CardFooter>
    </Card>
  );
};

export default PlainCard;
