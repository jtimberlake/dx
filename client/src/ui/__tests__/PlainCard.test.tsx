import React from 'react';
import { render } from '@testing-library/react';
import PlainCard from '../PlainCard';

const Plain = () => (
  <PlainCard title="Plain Card Title">
    <p>Can have any children inside</p>
    <span>Any element can work</span>
  </PlainCard>
);

test('Should render with children and match snapshot', () => {
  const { container } = render(<Plain />);
  expect(container).toMatchSnapshot();
});

test('Title in rendered element should be "Plain Card Title"', () => {
  const { getByText } = render(<Plain />);
  expect(getByText('Plain Card Title')).toBeInTheDocument();
});
