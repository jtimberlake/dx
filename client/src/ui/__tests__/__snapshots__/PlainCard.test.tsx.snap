// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render with children and match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 8px;
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 2px 1px -1px rgba(0,0,0,0.12);
  background-color: #ffffff;
  overflow: hidden;
  margin-bottom: 16px;
  padding: 16px;
}

.c1 {
  color: #7B746F;
  font-weight: normal;
  font-size: 1.8rem;
  margin: 0 0 16px 0;
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      Plain Card Title
    </h2>
    <p>
      Can have any children inside
    </p>
    <span>
      Any element can work
    </span>
  </div>
</div>
`;
